### ISSUE TITLE
Unable to change country for photos

### ISSUE THREAD

**Issue by O1d** [photoprism/photoprism#678]
Some of my photos get incorrectly located/tagged with a wrong country and I cannot change the country for these photos through the web interface

The photos in question were shot in Crimea, Ukraine before 2014, and while most of them are correctly identified as Ukraine, some have 'russian federation' location and label, which is wrong (even if one chooses to ignore the internationally recognized borders, there was no any 'russian federation' there before the occupation/annexation in 2014)

Though the main issue here is that photoprism doesn't allow to change the country in this case

While I'm gonna try to change it directly in the DB as a workaround, I believe there should be (at the very least) a possibility to change the country manually through edit menu

Thank you very much

**Comment by lastzero** [comment]
I see the problem. Our geo data has no historic context. While we might enable the country dropdown even when there are coordinates, your changes might get overwritten after the next update and won't have a global effect. This can be annoying.

**Comment by O1d** [comment]
But the [documentation](https://docs.photoprism.org/user-guide/organize/edit/) says that:

> Once you change data manually PhotoPrism always keeps your data even if you index again.

Isn't that the case?

**Comment by lastzero** [comment]
Yes, but not partially... like when you use coordinates, but want to manually specify the time zone, country, and everything else that's connected to it. We could do this next year, given enough funding. It would also increase the complexity of our app. Wondering how Google and others are solving this?

**Comment by O1d** [comment]
> Wondering how Google and others are solving this?

Well, Gphotos doesn't show country at all, just town/municipality location.

Though when I search for country names, it's just as inconsistent, most (if not all) photos can be found as Ukraine, but some (around 10% or so) are shown when I search russia

**Comment by lastzero** [comment]
> > Wondering how Google and others are solving this?
> 
> Well, Gphotos doesn't show country at all, just town/municipality location.

Pragmatic solution :laughing:

**Comment by Illia-M** [comment]
I am deeply concerned to see that this issue has remained unresolved for so long. Your software or its third-party integrations continue to display Ukrainian Crimea as part of russia, which is factually incorrect and misaligned with international standards.  

As Ukrainians, we are not only fighting and dying for the freedom and sovereignty of our own country but also for the freedom and democratic values of Europe as a whole. This is a fight for the principles that unite the free world. Yet, we also have to continue this fight in the digital world, where representation matters greatly. Software like yours plays a crucial role in showing whether it respects these values or unwittingly supports Russia’s illegal actions.  

Crimea is internationally recognized as part of Ukraine, and the European Union upholds a strict non-recognition policy regarding russia's illegal annexation of Crimea. You can read more about the EU's stance [here](https://www.consilium.europa.eu/en/press/press-releases/2024/06/17/russia-s-illegal-annexation-of-crimea-and-the-city-of-sevastopol-eu-extends-sanctions-for-further-year/).  

As an example, here is a screenshot from your software showing the incorrect representation of Ukrainian Crimea:  

![Incorrect map representation](https://github.com/user-attachments/assets/60a5d666-8cfa-4f49-a163-38016da48318)

I have spent a long time searching for a top-level photo management software solution for self-hosted purposes, and I was thrilled to find your software. I truly appreciate the features and capabilities it offers and would love to continue using it. However, this misunderstanding about Crimea's representation must be addressed to align with international standards and my personal values.  

I kindly urge your team to address this issue and update the software to reflect this correctly. Resolving this would demonstrate your commitment to freedom, truth, and compliance with international norms.  

Could you please provide an update on when this might be addressed?

**Comment by graciousgrey** [comment]
Thank you very much for your feedback! 

It seems that the underlying data has been corrected by openstreetmap in the meantime, so an update of our places service should fix this problem.

We will take care of it as soon as possible. You can follow the progress here: 
- https://github.com/photoprism/photoprism/issues/4688

**Comment by Illia-M** [comment]
Thanks so much for looking into this! I really appreciate it and can’t wait to see it resolved.

**Comment by lastzero** [comment]
@Illia-M Thanks again for your report! Since we do not have many (test) pictures from regions other than Europe, North America and southern Africa, glitches like this can go unnoticed - so we depend on your feedback to improve our services.

When location details are requested from our backend, it first checks the cache and the built-in database for a matching record, then does a reverse lookup with Photon, and only if that does not return a complete record, it uses our internal [Nominatim](https://nominatim.org/) service to complete the data.

While the [Photon](https://github.com/komoot/photon) database dumps available at http://download1.graphhopper.com/public/ should be based on OpenStreetMap (OSM), even their latest build combined with the latest dump does not return the same country as Nominatim (we are currently updating Nominatim to the very latest version to make sure this hasn't changed in the meantime, but the update will take a few days):

![photon-updated](https://github.com/user-attachments/assets/26fb915a-7ef6-4ac9-8fa6-45c638ac4fd4)

I would certainly be interested in learning more about this, however, we already have a lot of other work on our plate. For example, it could be that different data exists for each language/region, but that's pure speculation. 

Maybe you (or someone else) can check the available documentation, user forums, or ask the developers for clues?

- https://github.com/komoot/photon
- https://dl.photoprism.app/assets/osm/README

As a temporary fix, our backend will now always query Nominatim in addition to Photon when the country code is "ru". This will make uncached queries a bit slower for locations that Photon reports as being in Russia, so ideally we can remove this workaround again in the future:

![places-result](https://github.com/user-attachments/assets/b12b17a6-256c-4ad3-bb71-ddcd46d006ba)

Does this look alright to you?

**Comment by Illia-M** [comment]
Thank you so much for taking the time to investigate this issue in detail and for implementing a hotfix so quickly! I truly appreciate the effort you’ve put into addressing this, especially considering the complexity of handling location data across different services and datasets.

For Ukrainians, this fix is more than just a technical adjustment - it’s a symbol of support and acknowledgment of our sovereignty. It means a lot to see that even in the digital world, there’s care and effort to reflect the truth and respect international norms. This kind of action goes a long way in showing solidarity, and I’m incredibly grateful for that.

I understand the challenges you’ve highlighted and the workload your team is managing. I’ll do my best to contribute to this amazing open-source project, whether it’s by exploring the documentation, sharing feedback, or helping with testing where possible. It’s the least I can do to support a project that has shown such dedication to its community.

Thanks again for everything you’ve done—it’s truly appreciated by me and many others!

**Comment by lastzero** [comment]
@Illia-M The upgrade of our Nominatim instances has been completed with the following result:

**Old Location Data**

```json
{
  "id": "s2:40eaa18bd52c",
  "name": "МЕДУЗА",
  "street": "35К-004;Р-25",
  "postcode": "96505",
  "category": "water park",
  "timezone": "Europe/Simferopol",
  "lat": 45.151546,
  "lng": 33.493126,
  "place": {
    "id": "ua:24tNYzBzqJGE",
    "label": "Прибрежное, Республика Крым, Ukraine",
    "district": "",
    "city": "Прибрежное",
    "state": "Республика Крым",
    "country": "ua",
    "keywords": ""
  }
}
```

**New Location Data**

```json
{
  "id": "s2:40eaa18bd52c",
  "name": "Water Park",
  "street": "Евпаторийская улица",
  "postcode": "96505",
  "category": "water park",
  "timezone": "Europe/Simferopol",
  "lat": 45.151546,
  "lng": 33.493126,
  "place": {
    "id": "ua:24tNYzBzqJGE",
    "label": "Прибрежное, Республика Крым, Ukraine",
    "district": "",
    "city": "Прибрежное",
    "state": "Республика Крым",
    "country": "ua",
    "keywords": ""
  }
}
```

Does this look good to you?

**Comment by Illia-M** [comment]
@lastzero In both results, the country is Ukraine, so all the data is correct. I reviewed the changes in this area on OSM and noticed that, from time to time, someone changes it to the incorrect russian territory. This seems to be more of an issue with OSM’s moderation mechanics. I really appreciate your quick feedback and the actions you’ve taken. Thank you!

P.S. I’m continuing to explore this amazing tool, which is the first (ever!!) that can handle my old scanned library with raw image files ~4 GB each. It’s incredible! I’ll keep discovering more and contributing wherever I can.
